### PR TITLE
Doc updates for extensions

### PR DESCRIPTION
**Pull request by danmoseley**
From running the doc porting tool on all of extensions.

Fixes https://github.com/dotnet/runtime/issues/54870

### CODE REVIEW

**Comment by danmoseley** [comment]
I guess there's some fixups before we can merge this.
@gewarren what's the correct replacement for `<list><item><description>` etc? Not sure why they used that. I can changed to bulleted list using asterisks?

**Comment by maryamariyan** [comment]
@danmoseley were all the docs taken as-is from triple slash comments in dotnet/runtime repo or you also had to manually add docs for some?

File: xml/Microsoft.Extensions.Hosting/HostingHostBuilderExtensions.xml [hunk 1]
```diff
@@ -170,10 +206,10 @@
         <Parameter Name="configureOptions" Type="System.Action&lt;Microsoft.Extensions.Hosting.HostOptions&gt;" Index="1" FrameworkAlternate="dotnet-plat-ext-6.0" />
       </Parameters>
       <Docs>
-        <param name="hostBuilder">To be added.</param>
-        <param name="configureOptions">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="hostBuilder">The <see cref="T:Microsoft.Extensions.Hosting.IHostBuilder" /> to configure.</param>
+        <param name="configureOptions">The delegate for configuring the <see cref="T:Microsoft.Extensions.Hosting.HostOptions" />.</param>
+        <summary>Adds a delegate for configuring the <see cref="T:Microsoft.Extensions.Hosting.HostOptions" /> of the <see cref="T:Microsoft.Extensions.Hosting.IHost" /> instance related to th.</summary>
```
Comment (maryamariyan) [inline xml/Microsoft.Extensions.Hosting/HostingHostBuilderExtensions.xml:211]: this sentence is ending unexpectedly here

File: xml/Microsoft.Extensions.DependencyInjection/ServiceProviderServiceExtensions.xml [hunk 2]
```diff
@@ -70,9 +70,9 @@
         <Parameter Name="provider" Type="System.IServiceProvider" Index="0" FrameworkAlternate="dotnet-plat-ext-6.0" RefType="this" />
       </Parameters>
       <Docs>
-        <param name="provider">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="provider">The <see cref="T:System.IServiceProvider" /> to create the scope from.</param>
+        <summary>Creates a new <see cref="T:Microsoft.Extensions.DependencyInjection.AsyncServiceScope" /> that can be used to resolve scoped services.</summary>
+        <returns>A <see cref="T:Microsoft.Extensions.DependencyInjection.AsyncServiceScope" /> that can be used to resolve scoped services.</returns>
```
Comment (maryamariyan) [inline xml/Microsoft.Extensions.DependencyInjection/ServiceProviderServiceExtensions.xml:75]: perhaps?

```suggestion
        <returns>An <see cref="T:Microsoft.Extensions.DependencyInjection.AsyncServiceScope" /> that can be used to resolve scoped services.</returns>
```

File: xml/Microsoft.Extensions.DependencyInjection/AsyncServiceScope.xml [hunk 3]
```diff
@@ -30,7 +30,7 @@
     </Attribute>
   </Attributes>
   <Docs>
-    <summary>To be added.</summary>
+    <summary>A <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceScope" /> implementation that implements <see cref="T:System.IAsyncDisposable" />.</summary>
```
Comment (maryamariyan) [inline xml/Microsoft.Extensions.DependencyInjection/AsyncServiceScope.xml:33]: same thought here.

**Review by maryamariyan — APPROVED** [review]

**Comment by danmoseley** [comment]
> @danmoseley were all the docs taken as-is from triple slash comments in dotnet/runtime repo 

Yes. I didn't and don't want to edit anything here -- if any edits are needed, I want to make them in the repo, and regenerate. I will do so..

**Comment by gewarren** [comment]
> I guess there's some fixups before we can merge this.
> @gewarren what's the correct replacement for `<list><item><description>` etc? Not sure why they used that. I can changed to bulleted list using asterisks?

Yes, asterisks or hyphens will work.

File: xml/Microsoft.Extensions.DependencyInjection/AsyncServiceScope.xml [hunk 4]
```diff
@@ -51,7 +51,9 @@
       </Parameters>
       <Docs>
         <param name="serviceScope">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>Initializes a new instance of the <see cref="T:Microsoft.Extensions.DependencyInjection.AsyncServiceScope" /> struct.
+            Wraps an instance of <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceScope" />.
+            <param name="serviceScope">The <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceScope" /> instance to wrap.</param></summary>
```
Comment (gewarren) [inline xml/Microsoft.Extensions.DependencyInjection/AsyncServiceScope.xml:56]: Something weird happened here and the param is nested inside the summary.

Comment (gewarren) [inline xml/Microsoft.Extensions.DependencyInjection/AsyncServiceScope.xml:56]:
```suggestion
        </summary>
        <param name="serviceScope">The <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceScope" /> instance to wrap.</param>
```

**Review by gewarren — COMMENTED** [review]
Do you still plan to convert the bulleted list?

**Comment by danmoseley** [comment]
It should be good now. What was confusing me for a while is that the port tool will not port text into the docs unless the doc text is either blank or "To be added.". So if you run the tool locally, then change the text in the code/xml and run it again, it will do nothing. You have to reset all your docs changes before you run it again.

**Comment by danmoseley** [comment]
> Do you still plan to convert the bulleted list?

@gewarren I did?
https://github.com/dotnet/dotnet-api-docs/pull/6959/files#diff-2cfecbcaa097348c067fc3db0fdf246c843b35836e0b8c4a2d0301a3a78e491eR128

**Pull request merged**